deepinv/physics/generator/mri.py

```diff
@@ -1,83 +1,244 @@
+            self.C, self.T, self.H, self.W = self.img_size
+        else:
+            raise ValueError("img_size must be (H, W) or (C, H, W) or (C, T, H, W)")
 
-    def step(self, batch_size=1):
+        self.n_center = int(self.center_fraction * self.W)
+        self.n_lines = self.W // self.acc - self.n_center
+
+        if self.n_lines < 0:
+            raise ValueError(
+                "center_fraction is too high for this acceleration factor."
+            )
+        elif self.n_lines == 0:
+            warnings.warn(
+                "Number of high frequency lines to be sampled is 0. Reduce acceleration factor or reduce center_fraction."
+            )
+
+    def sample_mask(self, mask: torch.Tensor) -> torch.Tensor:
+        """Given empty mask, sample lines according to child class sampling strategy.
+
+        This must be implemented in child classes. Time dimension is specified but can be ignored if needed.
+
+        :param torch.Tensor mask: empty mask of shape (B, C, T, H, W)
+        :return torch.Tensor: sampled mask of shape (B, C, T, H, W)
+        """
+        raise NotImplementedError()
+
+    def step(self, batch_size=1) -> dict:
         r"""
         Create a mask of vertical lines.
 
         :param int batch_size: batch_size.
-        :return: dictionary with key ``mask``: tensor of size (batch_size, C, H, W) with values in {0, 1}.
+        :return: dictionary with key **'mask'**: tensor of size (batch_size, C, H, W) or (batch_size, C, T, H, W) with values in {0, 1}.
         :rtype: dict
         """
+        _T = self.T if self.T > 0 else 1
+        mask = self.sample_mask(
+            torch.zeros((batch_size, self.C, _T, self.H, self.W), **self.factory_kwargs)
+        )
 
-        if len(self.img_size) == 2:
-            H, W = self.img_size
-            C = 2
-        elif len(self.img_size) == 3:
-            C, H, W = self.img_size
-        else:
-            raise ValueError("img_size must be (C, H, W) or (H, W)")
-
-        acceleration_factor = self.acceleration
-
-        if acceleration_factor == 4:
-            central_lines_percent = 0.08
-            num_lines_center = int(central_lines_percent * W)
-            side_lines_percent = 0.25 - central_lines_percent
-            num_lines_side = int(side_lines_percent * W)
-        if acceleration_factor == 8:
-            central_lines_percent = 0.04
-            num_lines_center = int(central_lines_percent * W)
-            side_lines_percent = 0.125 - central_lines_percent
-            num_lines_side = int(side_lines_percent * W)
-
-        mask = torch.zeros((batch_size, H, W), **self.factory_kwargs)
-
-        center_line_indices = torch.linspace(
-            H // 2 - num_lines_center // 2,
-            H // 2 + num_lines_center // 2 + 1,
-            steps=50,
-            dtype=torch.long,
+        if self.T == 0:
+            mask = mask[:, :, 0, :, :]
+
+        return {"mask": mask}
+
+
+class RandomMaskGenerator(BaseMaskGenerator):
```

Comment:
Should we call this RandomUniformMaskGenerator? 

Or merge UniformMaskGenerator and RandomMaskGenerator into UniformMaskGenerator with a random=True/False option in the init?